<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FormListings</class>
 <widget class="QWidget" name="FormListings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1088</width>
    <height>756</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Repository Search and Listing</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qBorgPilot.qrc">
    <normaloff>:/images/borgpilot_16x16.png</normaloff>:/images/borgpilot_16x16.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBoxRepos">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>300</height>
      </size>
     </property>
     <property name="title">
      <string>Repositories</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLineEdit" name="lineEdit_ScriptName">
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="maxLength">
         <number>15</number>
        </property>
        <property name="placeholderText">
         <string>script name</string>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="TreeWidgetRepo_P" name="treeWidget_Repos">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::EditTrigger::DoubleClicked|QAbstractItemView::EditTrigger::EditKeyPressed|QAbstractItemView::EditTrigger::SelectedClicked</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
        </property>
        <column>
         <property name="text">
          <string notr="true">Repository</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>::Archive</string>
         </property>
        </column>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QToolButton" name="toolButton_Exec">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QToolBox" name="toolBox_SearchAndListing">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="page_Search">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1070</width>
        <height>368</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Search</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Pass Phrase</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEdit_PassPhrase">
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="maxLength">
          <number>20</number>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>468</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="3" rowspan="4">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Placeholders Description</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="TreeViewPatDesc_P" name="treeView_PatDesc">
            <property name="maximumSize">
             <size>
              <width>600</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::NoSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Pattern</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_Pattern">
           <property name="placeholderText">
            <string>/path/to/repo::archiveA --format=&quot;{mode} {user:6} {group:6} {size:8d} {isomtime} {path} {extra}{NEWLINE}&quot;</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButton_ExecSearch">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run Search&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0" colspan="3">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_Ex02_2">
           <property name="text">
            <string>e.g.: (just the repository name will list all associated archives) /storage/repo01</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_Ex01">
           <property name="text">
            <string>e.g.: /storage/repo01::project-2025-05-01 --format=&quot;{mode} {user:6} {group:6} {size:8d} {isomtime} {path}{extra}{NEWLINE}&quot;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_Ex02">
           <property name="text">
            <string>e.g.: /storage/repo01::project-2025-05-01 --pattern '+ re:.ext$' --pattern '- re:^.*$'</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>207</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_Listing">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1070</width>
        <height>368</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Results</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QToolButton" name="toolButtonSaveToFile">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save results to file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset theme="QIcon::ThemeIcon::DocumentSaveAs"/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTextEdit" name="textEdit_Results">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(0, 0, 0);
color: rgb(0, 85, 0);</string>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::LineWrapMode::NoWrap</enum>
         </property>
         <property name="lineWrapColumnOrWidth">
          <number>0</number>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::TextInteractionFlag::TextSelectableByKeyboard|Qt::TextInteractionFlag::TextSelectableByMouse</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreeViewPatDesc_P</class>
   <extends>QTreeView</extends>
   <header location="global">treeviewpatdesc_p.h</header>
  </customwidget>
  <customwidget>
   <class>TreeWidgetRepo_P</class>
   <extends>QTreeWidget</extends>
   <header location="global">treewidgetrepo_p.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="qBorgPilot.qrc"/>
 </resources>
 <connections/>
</ui>
